client/components/App.js

```diff
@@ -1,32 +1,61 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { HashRouter as Router, Route } from 'react-router-dom';
-import Home from './home/Home'
-import Build from './build/Build'
-import QuestionWalkthrough from './build/QuestionWalkthrough'
-import NavBar from './NavBar'
-import SignIn from './SignIn'
+import axios from 'axios';
+import Home from './home/Home';
+import Build from './build/Build';
+import QuestionWalkthrough from './build/QuestionWalkthrough';
+import NavBar from './NavBar';
+import SignIn from './SignIn';
 
-function App() {
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      user: false,
+    };
+    this.logout = this.logout.bind(this);
+  }
 
+  componentDidMount() {
+    // upon load, check if there is a loged in user
+    // in future, use cookies instead LS
+    const token = window.localStorage.getItem('token');
+    if (token) {
+      axios.get('/api/auth', {
+        headers: {
+          authorization: token,
+        },
+      });
+      this.setState({ user: true });
+    }
+  }
+
+  logout() {
+    window.localStorage.removeItem('token');
+    this.setState({ user: false });
+  }
+
```

Comment:
This logout function is also something we could also easily replace with a setState method through React hooks